### ISSUE TITLE
no-invalid-role false negative

### ISSUE THREAD

**Issue by boris-petrov** [ember-template-lint/ember-template-lint#1558]
`no-invalid-role` should give a warning in a case like this:

```hbs
{{some-component tagName="li" role="presentation"}}
```

This is a bit error-prone because `role` might not map to an attribute in the same tag but perhaps that's going to mostly be the case so a warning could be useful.

**Comment by MelSumner** [comment]
Good catch!

**Comment by rajasegar** [comment]
Need more clarity on the issue, are we going to throw warnings for all the `Mustache` statements with role attribute or do we need to check the if there is a `tagName` attribute present? At preset we are checking only for `Element` nodes.

**Comment by boris-petrov** [comment]
@rajasegar - well, I believe `tagName` should also be checked because for some tags the `role` attribute is totally correct but for others (like the example I've given with `li`) it is not.

**Comment by MelSumner** [comment]
@rajasegar I think we can check inside mustache statements for the presence of both tagname and role and if both are present, check the values that they are set to. If they are dynamic then that will have to be okay, but then if they are not, we can check them to make sure they're not an invalid combination.
